google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import java.nio.file.Path;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * {@link CloudSdkService} providing SDK that is managed and automatically installable, see {@link
+ * com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk}
+ */
+// TODO(ivanporty) implementation coming in the next PR
+public class ManagedCloudSdkService implements CloudSdkService {
+
+  @Nullable
+  @Override
+  public Path getSdkHomePath() {
+    return null;
+  }
+
+  @Override
+  public void setSdkHomePath(String path) {}
+
+  @Override
+  public SdkStatus getStatus() {
```

Comment:
just to get a better idea: how do you intend for this SdkStatus to be consumed? I'm curious about how logic that needs to wait for the sdk to become available (e.g. deployment) will use this.

Reply:
Mostly for the uncompleted /  failed managed SDK installs. When we do an action that require SDK, we then check for `SdkStatus`. Roughly similar to:
* INVALID, NOT_AVAILABLE - attempt `install()` and see if it's supported, if not, suggest settings (for custom SDK case).
* INSTALLING - wait until ready, attach listener, wait for status change to `READY`. If status changes to `INVALID`, suggests checking logs for errors, attempt again.

Reply:
aha I see. sounds good